code/modules/mob/living/living_healthscan.dm

```diff
@@ -0,0 +1,655 @@
+					advice += list(list(
+						"advice" = "Administer a single dose of peridaxon.",
+						"icon" = "syringe",
+						"colour" = "grey"
+						))
+				if(human_target_mob.getOxyLoss() > 50 && chemicals_lists["dexalin"] < 5)
+					advice += list(list(
+						"advice" = "Administer a single dose of dexalin.",
+						"icon" = "syringe",
+						"colour" = "blue"
+						))
+				if(human_target_mob.getFireLoss(organic_only = TRUE) > 30 && (chemicals_lists["kelotane"] < 3 || chemicals_lists["dermaline"] < 3))
+					advice += list(list(
+						"advice" = "Administer a single dose of kelotane.",
+						"icon" = "syringe",
+						"colour" = "yellow"
+						))
+				if(human_target_mob.getBruteLoss(organic_only = TRUE) > 30 && (chemicals_lists["bicaridine"] < 3 || chemicals_lists["meralyne"] < 3))
+					advice += list(list(
+						"advice" = "Administer a single dose of bicardine.",
+						"icon" = "syringe",
+						"colour" = "red"
+						))
+				if(human_target_mob.health < 0 && chemicals_lists["inaprovaline"] < 5)
+					advice += list(list(
+						"advice" = "Administer a single dose of inaprovaline.",
+						"icon" = "syringe",
+						"colour" = "purple"
+						))
+
+				var/has_pain = FALSE
+				for(var/datum/effects/pain/P in target_mob.effects_list)
+					has_pain = TRUE
+					break
+
+				if(has_pain && chemicals_lists["tramadol"] < 3 && !chemicals_lists["paracetamol"])
+					advice += list(list(
+						"advice" = "Administer a single dose of tramadol.",
+						"icon" = "syringe",
+						"colour" = "white"
+						))
+				if(chemicals_lists["paracetamol"])
+					advice += list(list(
+						"advice" = "Do NOT administer tramadol.",
+						"icon" = "window-close",
+						"colour" = "red"
+						))
+
+			if(advice.len)
+				data["advice"] = advice
+
+		//diseases
+		var/list/diseases = list()
+		for(var/datum/disease/disease in target_mob.viruses)
+			if(!disease.hidden[SCANNER] || detail_level >= DETAIL_LEVEL_FULL)
+				var/current_disease = list(
+					"name" = disease.name,
+					"form" = disease.form,
+					"type" = disease.spread,
+					"stage" = disease.stage,
+					"max_stage" = disease.max_stages,
+					"cure" = disease.cure
+				)
+				diseases += list(current_disease)
+		if(diseases.len)
+			data["diseases"] = diseases
+
+	if(target_mob.getBrainLoss() >= 100 || !target_mob.has_brain())
+		data["ssd"] = "Subject is brain-dead."
+	else if(target_mob.has_brain() && target_mob.stat != DEAD && ishuman(target_mob))
+		if(!target_mob.key)
+			data["ssd"] = "No soul detected." // they ghosted
+		else if(!target_mob.client)
+			data["ssd"] = "SSD detected." // SSD
+
+	return data
+
+/// legacy proc for to_chat messages on health analysers
+/mob/living/proc/health_scan(mob/living/carbon/human/user, var/ignore_delay = FALSE, var/mode = 1, var/hud_mode = 1, var/alien = FALSE, var/do_checks = TRUE) // ahem. FUCK WHOEVER CODED THIS SHIT AS NUMBERS AND NOT DEFINES.
```

Comment:
Would it be worth reworking this to go off defines rather than numbers?

Reply:
legacy code = ick but I'll do it